javascript/javascript1/week3/warmup3.js

```diff
@@ -0,0 +1,39 @@
+const seriesDurations = [
+  {
+    title: "Game of thrones",
+    days: 3,
+    hours: 1,
+    minutes: 0,
+  },
+  {
+    title: "Sherlock",
+    days: 0,
+    hours: 18,
+    minutes: 0,
+  },
+  {
+    title: "Vikings",
+    days: 4,
+    hours: 13,
+    minutes: 0,
+  },
+];
+const myAge = 29;
+
+function calculateLifeSpentOnSeries(seriesArray, age){
+  const ageInHour = age * 365.25 * 24;
+  const result = [];
+
+  for (let i = 0; i<seriesArray.length; i++){
+    const seriesInHour = seriesArray[i].days * 24 + seriesArray[i].hours;
+    const percentage = (seriesInHour / ageInHour *100).toFixed(4);
+    result.push(`${seriesArray[i].title} took ${percentage}% of my life`)
```

Comment:
Can you think of a way to simplify this and the printing logic below? 🔍 I assume your motivation for putting it into an array is to have line breaks; if so, can you think of another way to do so? 😄

Reply:
Thanks for the feedback! 😊 I was considering using console.log inside the loop, but from what I understand, returning the result is a better approach. I'm not entirely sure, though. Could you help me understand when it's better to use return versus console.log?

Reply:
It is very much dependent on what the intention of the function is. 🙂  `console.log` is often seen as a "code smell" (aka "you're probably doing something weird" if you have it in your code), so you will almost always want to stick with returning stuff; this also gives the most flexibility to users of your function. E.g., in this context, the caller might want to show the text in the console or maybe put it into the HTML. So returning the message here is perfectly fine! 😄 

If you want to stick with returning here and avoid the array, then you could use a string which you concatenate to and use `\n` to make line breaks. Something like this:
```js
function calculateLifeSpentOnSeries(seriesArray, age) {
  let message = "";
  // ...
  for (let i = 0; i < seriesArray.length; i++) {
    // ...
    message += `${seriesArray[i].title} took ${percentage}% of my life\n`;
  }
  return message;
}
```

This is a bit of a contrived example and will also result in a loose `\n` in the end which is not super nice. Generally, such a function as this would not make any assumptions about the formatting but instead return structured data such as an object and then the caller can decide what to do. 😄